privacyidea/lib/monitoringstats.py

```diff
@@ -27,18 +27,33 @@
 This module is tested in tests/test_lib_monitoringstats.py
 """
 import logging
-from dateutil.tz import tzlocal, tzutc
-import traceback
+from dateutil.tz import tzlocal
 from privacyidea.lib.log import log_with
-from privacyidea.lib.utils import convert_timestamp_to_utc
-from privacyidea.models import MonitoringStats
-from sqlalchemy import and_, distinct
-from privacyidea.lib.tokenclass import AUTH_DATE_FORMAT
+from privacyidea.lib.utils import convert_timestamp_to_utc, get_module_class
+from privacyidea.lib.framework import get_app_config
 import datetime
 
 log = logging.getLogger(__name__)
 
 
+@log_with(log, log_entry=False)
+def _get_monitoring():
+    """
+    This wrapper function creates a new monitoring object based on the config
+    from the config file. The config file entry could look like this:
+
+        PI_MONITORING_MODULE = privacyidea.lib.monitoringmodule.sqlstats
+
+    Each monitoring module can have its own config values.
+
+    :return: Monitoring Object
+    """
+    config = get_app_config()
+    monitoring_module = config.get("PI_MONITORING_MODULE", "privacyidea.lib.monitoringmodules.sqlstats")
+    monitoring = get_module_class(monitoring_module, "Monitoring")(config)
```

Comment:
If we have three event handlers writing statistics in one request, we will create three ``MonitoringModule`` objects in that request. Should we maybe cache a monitoring module in the request-local store, similarly to the audit object?

Reply:
This probably depends on whether the backend likes being cached or reuse connections or not.
I think we should offer this but make this configurable. I would like to put this into another issue.

Reply:
will do 
from privacyidea.lib.framework import request_local_store
See audit and and resolver

Reply:
done in 315c309

Reply:
:+1: